### ISSUE TITLE
Move directives decorator to operations.directive

### ISSUE THREAD

**Issue by vyasr** [glotzerlab/signac-flow#309]
### Feature description

It would be nice if groups and operations were unified to a great degree. One area of cognitive dissonance is that directives are currently assigned as a distinct decorator for operations, but are part of the groups namespace, i.e.
```
@FlowProject.operation
@group.directives(...)
@directives(...)
def op(job):
    pass
```

### Proposed solution

We should consider moving this decorator into the operation namespace, enabling something more symmetric like:

```
@FlowProject.operation(...)
@group.directives(...)
def op(job):
    pass
```

**Comment by b-butler** [comment]
Currently the APi for `FlowGroupEntry` is `group.with_directives`. I propose we use that instead of `operation.directive`

**Comment by csadorf** [comment]
@vyasr I don't fully follow your example, is the idea to provide to directives as part of the `FlowOperation` constructor?

**Comment by vyasr** [comment]
@csadorf yes exactly, in the same way that the API works for groups.

**Comment by csadorf** [comment]
Ok, although I like @b-butler 's proposal.

**Comment by bdice** [comment]
I agree that @b-butler's suggestion would make for a pretty and intuitive API. To make things more concrete for future readers / further discussion:

```python
@TestProject.operation
@flow.directives(nranks=4, omp_num_threads=2)
def hybrid_op(job):
    pass
```

would become one of the following:

```python
# Vyas's proposal: fewer words required.

@TestProject.operation(nranks=4, omp_num_threads=2)
def hybrid_op(job):
    pass

# Brandon's proposal: match the API of group.with_directives.

@TestProject.operation.with_directives(nranks=4, omp_num_threads=4)
def hybrid_op(job):
    pass
```

**Comment by csadorf** [comment]
I'm in favor of Brandon's proposal! 👍

**Comment by bdice** [comment]
Resolved by #502.
